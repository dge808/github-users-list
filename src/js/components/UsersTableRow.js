import React from 'react';
import { hashHistory } from 'react-router';

const UsersTableRow = ({ rowInformation }) => {
    const handleRowClick = () => {
        hashHistory.push(`/users/${rowInformation.login}`)
    };

    return (
        <tr onClick={handleRowClick}>
            <td class="avatar-cell">
                <img src={rowInformation.avatar_url} class="user-avatar" alt={rowInformation.login}/>
            </td>
            <td style={{width: '35%'}}>
                {rowInformation.login}
            </td>
            <td style={{width: '35%'}}>
                {rowInformation.html_url}
            </td>
        </tr>
    )
};

export default UsersTableRow;

UsersTableRow.propTypes = {
    rowInformation: React.PropTypes.object
};
