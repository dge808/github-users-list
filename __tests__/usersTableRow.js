import React from 'react';
import renderer from 'react-test-renderer';
import UsersTableRow from '../src/js/components/UsersTableRow';

const rowInformation = {
    name: 'Alina',
    followers: 2,
    following: 2,
    created_at: "2015-07-31T05:24:19Z",
    company: null,
    email: "dge808@gmail.com",
    location: "Ukraine",
    blog: "",
    bio: null
};

describe('User Table Row(snapshot)', () => {
    it('should match Users Table Row snapshot', () => {
        const component = renderer.create(<UsersTableRow rowInformation={rowInformation}/>);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    })
});