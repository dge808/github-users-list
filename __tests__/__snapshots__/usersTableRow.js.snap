exports[`User Table Row(snapshot) should match Users Table Row snapshot 1`] = `
<tr
  onClick={[Function]}>
  <td
    className="avatar-cell">
    <img
      alt={undefined}
      className="user-avatar"
      src={undefined} />
  </td>
  <td
    style={
      Object {
        "width": "35%",
      }
    } />
  <td
    style={
      Object {
        "width": "35%",
      }
    } />
</tr>
`;
